### ISSUE TITLE
Renaming type in maven project does not work

### ISSUE THREAD

**Issue by GoogleCodeExporter** [bwebb3856/qwickie#34]
```
What steps will reproduce the problem?
1. Create a wicket project using maven
2. Create wicket component, including html in the resources source folder
3. Rename the java class.

What is the expected output? What do you see instead?
Expected is that the HTML file is renamed as well.
Instead a new HTML file is created in the same folder the java file resides.

What version of the product are you using? On what operating system?
0.9.3 on Ubuntu 10.10, Eclipse Indigo


Please provide any additional information below.
The problem I think is in QWickieTypeRenameParticipant, line 90.
the expression (ff.getParent().equals(oldFile.getParent())) return false, 
because the files ar ein different containers.

I've attached a patch which supplies a new method in TypeHelper (maybe not an 
ideal location) which looks if two resources are in the same relative path to 
their respective source folders. This method is called instead of the 
expression above. That seems to work OK, at least for me :)

I do want to take this opportunity to thank you for your work in creating 
qwickie, since it will be a part of my standard wicket/eclipse toolkit from now 
on!

```

Original issue reported on code.google.com by `w.f.vo...@gmail.com` on 18 Jan 2012 at 2:39

Attachments:
- [qwickie-renameType.patch](https://storage.googleapis.com/google-code-attachments/qwickie/issue-34/comment-0/qwickie-renameType.patch)

**Comment by GoogleCodeExporter** [comment]
```
Oh wow! Thats the best defect report I've had since the beginning of qwickie.
So many thanks for that (you must have read the whole code tinkering to get 
this fix).

I've tried it before: Fails. After your patch: Working!

So what I do is: I'll take your patch, commit it and release it on weekend when 
I've tested some more. Which name should I use for the credits?
```

Original comment by `count.negative` on 18 Jan 2012 at 3:05
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
Wow, That is fast :)

My name is Willem Voogd (wfvoogd@gmail.com)

My plan was to write a plugin like yours, but then I found qwickie, and
since I never wrote a plugin before, I welcomed this very much. We are
planning to make it part of our company wide eclipse stack, so therefore I
checked it out yesterday, the main use case (switching to and from
java<->html) works very well, so my advise to the guys who take the
decision is to go for it. I'm planning on some more improvements (support
for different language templates, refactoring also to change the properties
file name(s))

I am also posting a review as blog, I'll mail you the location when it's
there, since there are some things that I did not find, or found to be
defect, or maybe just did not understand correctly ;) , but I still have to
work it out in repro scenario's, but maybe you could comment on things I
missed in the blog.

Thanks for the great product!

Willem Voogd
```

Original comment by `w.f.vo...@gmail.com` on 18 Jan 2012 at 3:23

**Comment by GoogleCodeExporter** [comment]
```
Ha! That's exactly the reason I wrote qwickie. So the plugin code is not the 
best work I've done and learning the eclipse api was (and is) a pain.
There where many features I'd like to implement (maybe use wicket itself for 
evaluation) but as life goes, I'm running out of time. 

I'm hardly waiting for your further improvements to qwickie!

Thank you Willem, you're welcome.

count
```

Original comment by `count.negative` on 18 Jan 2012 at 3:40

**Comment by GoogleCodeExporter** [comment]
```
I think the haveSameRelativePathToParentSourceFolder method can be used for 
other (main) file searches (html, java, properties) as well. Since the 
filesearch in other places than the same folder as the java directory now only 
validates the filename as far as I know, theoretically a html file for a class 
in a different package could be found, when you have implemented the same 
classname in multiple packages. This new method should only return true if the 
directory stuctures match.

Which does lead me to another point, at our company, we have the page templates 
in a different directory completely, so it might be an idea to have a 
preferences form where you can override the standard places for (some) 
components.

I will look into the latter issue, since that affects our case immediately...
```

Original comment by `willem.v...@dayon.nl` on 19 Jan 2012 at 10:12

**Comment by GoogleCodeExporter** [comment]
```
In a complete different directory? Hm, so how does wicket know about them 
(pom.xml/web.xml/own ResourceStreamLocator in Application?)
Maybe qwickie can find the location by itself (convention over config).
But if not, there is a QWickieProjectPreferencePage where we can store the per 
project settings.


```

Original comment by `count.negative` on 19 Jan 2012 at 10:20

**Comment by GoogleCodeExporter** [comment]
```
Well it is within a source directory, but not the same relative path. I'm not 
sure, but i think the config is done programatically (A result of a project 
template which I have never really had a good look at), so it is not easy for 
qwickie to find it, if possible at all. Maybe once the usage of the wicket 
engine itself is used, it  is fixed automagically :-) But I wil have a look, 
maybe not today, but soon ;)
```

Original comment by `willem.v...@dayon.nl` on 19 Jan 2012 at 10:29

**Comment by GoogleCodeExporter** [comment]
```
It's an own ResourceStreamLocator.
```

Original comment by `willem.v...@dayon.nl` on 19 Jan 2012 at 3:30

**Comment by GoogleCodeExporter** [comment]
```
Released 0.9.4
```

Original comment by `count.negative` on 21 Jan 2012 at 8:40
- Changed state: **Fixed**
